assets/src/models/createDetourMachine.ts

```diff
@@ -24,10 +73,120 @@ export const createDetourMachine = setup({
   initial: "Detour Drawing",
   states: {
     "Detour Drawing": {
-      initial: "Editing",
+      initial: "Begin",
       states: {
+        Begin: {
+          always: [
+            {
+              guard: ({ context }) =>
+                context.routePattern !== undefined &&
+                context.route !== undefined,
+              target: "Editing",
+            },
+            { target: "Pick Route Pattern" },
+          ],
+        },
+        "Pick Route Pattern": {
+          initial: "Pick Route ID",
+          on: {
+            "detour.route-pattern.select-route": {
+              target: ".Pick Route ID",
+              actions: assign({
+                route: ({ event }) => event.route,
+              }),
+            },
+            /** This seems like it's needed, since it's possible to select
+             * ["Select a route"](https://github.com/mbta/skate/blame/e14d100c1db7df8bbf3c24ca4f8dbbe100d24d1e/assets/src/components/detours/detourRouteSelectionPanel.tsx#L68)
+             * But the `select` is "controlled"(in react terms),
+             * then we need a way to represent that state?
+             * Otherwise there's no way to see the "select a route" help message
+             *
+             * There is not yet a test for this because this is not a project requirement (yet. tbd).
+             * This also depends on the combobox implementation decisions.
```

Comment:
> The current `FormSelect` implementation has an entry with a `null` route ID also (I think), which should also trigger this event.

Yes, and just for clarity, I've explicitly handled that state instead of ignoring it https://github.com/mbta/skate/pull/2652/files#diff-a5c2ebb8d72dc32209d3598dadecd6793515e43155078116e8176cc292c32a0aR162-R173


---

I'll go ahead and edit this comment to remove the project requirement bit and probably the test bit?

Reply:
I'm deciding to remove the comment altogether, but if you disagree, lets chat about what to put instead.